### ISSUE TITLE
[FEATURE REQUEST] Multi-fold splits or random splits in NodeClassification tasks

### ISSUE THREAD

**Issue by GreatSnoopyMe** [Graph-Learning-Benchmarks/gli#73]
**Is your feature request related to a problem? Please describe.**

When writing `train_set`, `val_set` and `test_set` for the dataset, i found that not all the datasets has only one split of training set, validation set and test set. For `cora`, there is only one split, therefore the shape of its `graph.ndata['train_mask']` is `[2708]`. However, for `actor`, there are 10 splits of the set, therefore its `graph.train_mask` shape is `[7600,10]`. Another case is that, in the non-homophilous datasets like `arxiv-year`, its split style is by setting the split ratio and then randomly assign nodes into one of three sets (for `nodaclassification task`).

**Describe the solution you'd like**

After the meeting with Jiaqi, Jiaqi put forward some solution to it.
For the dataset like `cora`, which has only one split style of datasets, everything keeps the same.

For the dataset like `actor`, which has multiple ways of splitting datasets, we can add one key named `num_fold` to the `task.json` to denote there are `num_fold` ways to split the dataset (10 for the actor dataset). Further, we can change the key of `train_set`, `val_set` and `test_set` into `train_FOLD`, `val_FOLD` and `test_FOLD`. In the `actor.ipynb`, we can change the `dict`: `task_data` into 
```
task_data = {
    "train_0": train_set,
    "val_0": val_set,
    "test_0": test_set，
    "train_1": train_set,
    "val_1": val_set,
    "test_1": test_set，
    ......
    "train_9": train_set,
    "val_9": val_set,
    "test_9": test_set，
}
```
30 values in total. 

For the dataset like `arxiv-year`, which has no fixed way of splitting datasets but only randomly generate the splits with some pre-set ratio, we can delete the `train_set`, `val_set` and `test_set` in task.json  and replace them with `train_ratio`, `val_ratio` and `test_ratio`.

**Additional Context**

According to Jiaqi, the `dataloader` will need to adapt such change.
